src/platform_impl/apple/uikit/app_state.rs

```diff
@@ -42,25 +44,45 @@ macro_rules! bug_assert {
     };
 }
 
-pub(crate) struct EventLoopHandler {
-    #[allow(clippy::type_complexity)]
-    pub(crate) handler: Box<dyn FnMut(Event, &dyn RootActiveEventLoop)>,
-    pub(crate) event_loop: ActiveEventLoop,
-}
+/// Get the global event handler for the application.
+///
+/// This is stored separately from AppState, since AppState needs to be accessible while the handler
+/// is executing.
+fn get_handler(mtm: MainThreadMarker) -> &'static EventHandler {
+    // TODO(madsmtm): Use `MainThreadBound` once that is possible in `static`s.
+    struct StaticMainThreadBound<T>(T);
 
-impl fmt::Debug for EventLoopHandler {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("EventLoopHandler")
-            .field("handler", &"...")
-            .field("event_loop", &self.event_loop)
-            .finish()
+    impl<T> StaticMainThreadBound<T> {
+        const fn get(&self, _mtm: MainThreadMarker) -> &T {
+            &self.0
+        }
     }
+
+    unsafe impl<T> Send for StaticMainThreadBound<T> {}
+    unsafe impl<T> Sync for StaticMainThreadBound<T> {}
+
+    // SAFETY: Creating `StaticMainThreadBound` in a `const` context, where there is no concept
+    // of the main thread.
+    static GLOBAL: StaticMainThreadBound<OnceCell<EventHandler>> =
+        StaticMainThreadBound(OnceCell::new());
```

Comment:
Remembering our talk about not storing anything in static memory.
No clue what the story here is exactly.

Reply:
I think I would still like to achieve that at _some_ point, but it's going to be quite difficult with the event handler (especially so on AppKit, where we have to submit events from inside `-[NSApplication sendEvent:]`, without the ability to store anything there, since I plan to swizzle (replace) the method to avoid requiring `WinitApplication` being registered as the main application).

The `EventHandler` is effectively moved from `AppState`, which is already stored in a `static`, so this PR doesn't really change the status quo (we just have two smaller statics now, instead of one big one).